### PR TITLE
fix(p2p): couple bitswap client and server

### PR DESCRIPTION
**Pull request by walldiss**
Decoupled client was not able to notify server on newly available blocks.

### CODE REVIEW

File: nodebuilder/p2p/bitswap.go [hunk 1]
```diff
@@ -32,36 +31,27 @@ const (
...
+		bitswap.ProvideEnabled(false), // we don't provide blocks over DHT
 		// NOTE: These below are required for our protocol to work reliably.
 		// // See https://github.com/celestiaorg/celestia-node/issues/732
-		server.SetSendDontHaves(false),
-	)
+		bitswap.SetSendDontHaves(false),
+	}
+	clOpts := []bitswap.Option{
+		bitswap.SetSimulateDontHavesOnTimeout(false),
+		bitswap.WithoutDuplicatedBlockStats(),
```
Comment (Wondertan) [inline nodebuilder/p2p/bitswap.go:43]: You could also add them in a single var and separate client server via comments

**Review by Wondertan — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

File: nodebuilder/p2p/bitswap.go [hunk 2]
```diff
@@ -32,36 +29,26 @@ const (
...
 		// // See https://github.com/celestiaorg/celestia-node/issues/732
-		server.SetSendDontHaves(false),
-	)
+		bitswap.SetSendDontHaves(false),
 
-	clnt := client.New(
-		params.Ctx,
-		net,
-		params.Bs,
-		client.WithBlockReceivedNotifier(srvr),
```
Comment (renaynay) [inline nodebuilder/p2p/bitswap.go:49]: so bc they're coupled again now, we don't need to specify this?

**Review by renaynay — DISMISSED** [review]
utack

File: nodebuilder/p2p/bitswap.go (hunk 2 above)
Comment (walldiss) [reply nodebuilder/p2p/bitswap.go:49]: yes. `client.WithBlockReceivedNotifier` is enabled inside bitswap.New constructor.

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**